src/lib/y2network/sysconfig/connection_config_writer.rb

```diff
@@ -18,30 +18,40 @@
 # find current contact information at www.suse.com.
 
 require "y2network/sysconfig/interface_file"
+require "y2network/sysconfig/routes_file"
+
+Yast.import "Host"
 
 module Y2Network
   module Sysconfig
+    # This class is responsible for writing interfaces changes
     class ConnectionConfigWriter
       include Yast::Logger
 
       # Writes connection config to the underlying system
       #
+      # The method can receive the old configuration in order to perform clean-up tasks.
+      #
       # @param conn [Y2Network::ConnectionConfig::Base] Connection configuration to write
-      def write(conn)
+      # @param old_conn [Y2Network::ConnectionConfig::Base,nil] Connection configuration to write
+      def write(conn, old_conn = nil)
+        return if conn == old_conn
         file = Y2Network::Sysconfig::InterfaceFile.new(conn.interface)
         handler_class = find_handler_class(conn.type)
         return nil if handler_class.nil?
+        remove(old_conn) if old_conn
         file.clean
         handler_class.new(file).write(conn)
         file.save
       end
 
       # Removes connection config from the underlying system
       #
-      # @param name [String] Connection name to remove
-      def remove(name)
-        ifcfg = Y2Network::Sysconfig::InterfaceFile.find(name)
+      # @param conn [Y2Network::Conn] Connection name to remove
+      def remove(conn)
+        ifcfg = Y2Network::Sysconfig::InterfaceFile.find(conn.interface)
         ifcfg && ifcfg.remove
+        Yast::Host.remove_ip(conn.ip.address.address.to_s) if conn.ip
```

Comment:
what about removal of aliases here also?

Reply:
Do you mean IP aliases? It is done automatically when writing the file.